Add more links to 24.1 Feature Highlights, standardize internal URLs (#18576)
cockroachdb/docs: src/current/_includes/releases/v24.1/v24.1.0.md

    <tr>
        <td>
            <p class="feature-summary">Physical Cluster Replication is now generally available</p>

>             <p class="feature-description"><a href="https://www.cockroachlabs.com/docs/v24.1/physical-cluster-replication-overview">Physical Cluster Replication</a> is an asynchonous replication feature that allows your cluster to recover from full-cluster failure with a low RPO and RTO. In 24.1, it is a GA feature, requiring an Enterprise license, and only available for self-hosted CockroachDB deployments. For more information, refer to the <a href="https://www.cockroachlabs.com/docs/v24.1/physical-cluster-replication-overview">docs</a> and <a href="https://www.cockroachlabs.com/blog/232-physical-cluster-replication/">blog post</a>.</p>

Reviewer: ```suggestion
            <p class="feature-description"><a href="https://www.cockroachlabs.com/docs/v24.1/physical-cluster-replication-overview">Physical Cluster Replication</a> is an asynchonous replication feature that allows your cluster to recover from full-cluster failure with a low RPO and RTO. In 24.1, it is a GA feature, requiring an Enterprise license, and only available for self-hosted CockroachDB deployments. For more information, refer to the <a href="https://www.cockroachlabs.com/docs/v24.1/physical-cluster-replication-overview">Physical Cluster Replication overview</a> and a <a href="https://www.cockroachlabs.com/blog/232-physical-cluster-replication/">blog post about Physical Cluster Replication</a>.</p>
```
Informative link text, "docs" is pretty informal.

    <tr>
        <td>
            <p class="feature-summary">Private connectivity in CockroachDB Dedicated with Google Cloud Private Service Connect</p>

>             <p class="feature-description"><a href="https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster.html#establish-private-connectivity">Private connectivity</a> using Private Service Connect is now available in preview for CockroachDB Dedicated clusters on GCP. For more information, refer to the <a href="https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster.html#establish-private-connectivity">docs</a> and <a href="https://youtu.be/lcPnFEv0_eE">demo video</a>.</p>

Reviewer: ```suggestion
            <p class="feature-description"><a href="https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster.html#establish-private-connectivity">Private connectivity</a> using Private Service Connect is now available in preview for CockroachDB Dedicated clusters on GCP. For more information, refer to <a href="https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster.html#establish-private-connectivity">Establish Private Connectivity</a> and a <a href="https://youtu.be/lcPnFEv0_eE">video about how to set it up</a>.</p>
```

        <td>
            <p class="feature-summary">Read Committed Isolation in GA</p>
            <p class="feature-description">Read Committed isolation is now GA. Under <code><a href="https://www.cockroachlabs.com/docs/v24.1/read-committed">READ COMMITTED</a></code> isolation, DDL operations can be auto-committed, and <code>RETRY_SERIALIZABLE</code> errors are absent. This offers a simpler migration path for applications and removes the need for client-side retry handling, which is necessary under <code>SERIALIZABLE</code> isolation.</p>

>             <p class="feature-description">For more information, you can also view a <a href="https://youtu.be/RX5SYAap57s">video demo</a> and the blog post "<a href="https://www.cockroachlabs.com/blog/232-read-committed-no-more-anomaly-tables/">Isolation levels without the anomaly table</a>".</p>

Reviewer: ```suggestion
            <p class="feature-description">For more information, you can also view a <a href="https://youtu.be/RX5SYAap57s">video about using Read Committed isolation</a> and the blog post <a href="https://www.cockroachlabs.com/blog/232-read-committed-no-more-anomaly-tables/">Isolation levels without the anomaly table</a>.</p>
```
I don't think we generally put link text in quotes.

    <tr>
        <td>
            <p class="feature-summary">SELECT FOR SHARE updates</p>

>             <p class="feature-description">A shared lock that is acquired explicitly using <a href="https://www.cockroachlabs.com/docs/v24.1/select-for-update#for-share-usage">SELECT FOR SHARE</a> or implicitly by a read-committed transaction, can now be re-acquired with higher strength by either using a SELECT FOR UPDATE statement or by writing to the key</p>

Reviewer: ```suggestion
            <p class="feature-description">A shared lock that is acquired explicitly using <a href="https://www.cockroachlabs.com/docs/v24.1/select-for-update#for-share-usage"><code>SELECT FOR SHARE</code></a> or implicitly by a read-committed transaction, can now be re-acquired with higher strength by either using a <code>SELECT FOR UPDATE</code> statement or by writing to the key.</p>
```
I didn't check all of them, but these are long enough that they should end in a period.

    <tr>
        <td>
            <p class="feature-summary">SHOW DEFAULT SESSION VARIABLES FOR ROLE</p>

>             <p class="feature-description">Previously, if a database administrator wanted to know which session variables they had set by default for a role, they would need to switch into that user or scan a system table manually. This could be more time consuming or difficult to use. To address this, we have added a new command: <a href="https://www.cockroachlabs.com/docs/v24.1/show-default-session-variables-for-role">SHOW DEFAULT SESSION VARIABLES FOR ROLE</a>, which allows the database administrator to easily view the default values for session variables applied to a given user.</p>

Reviewer: ```suggestion
            <p class="feature-description">Previously, a database administrator could discover which session variables were set for a role only by switching into to that user user or querying a system table. The new command <a href="https://www.cockroachlabs.com/docs/v24.1/show-default-session-variables-for-role"><code>SHOW DEFAULT SESSION VARIABLES FOR ROLE</code></a> shows the database administrator the default values for session variables applied to a given user.</p>
```

- `sql.txn.read_committed_isolation.enabled` is now `true` by default. When set to `true`, transactions use the `READ COMMITTED` isolation level if specified by `BEGIN`/`SET` commands.
  - If the cluster setting is `false`, as was the default in v23.2, such `READ COMMITTED` transactions will instead run as `SERIALIZABLE`.
  - To check whether any transactions are being upgraded to `SERIALIZABLE`, see the [**Upgrades of SQL Transaction Isolation Level**](https://www.cockroachlabs.com/docs/v24.1/ui-sql-dashboard#upgrades-of-sql-transaction-isolation-level) graph in the DB Console."

> - `changefeed.balance_range_distribution.enable` is now deprecated. Instead, use the new [cluster setting]({% link v24.1/cluster-settings.md %}) `changefeed.default_range_distribution_strategy`. `changefeed.default_range_distribution_strategy='balanced_simple'` has the same effect as setting `changefeed.balance_range_distribution.enable=true`. It does not require `initial_scan='only'`, which was required by the old setting. [#115166][#115166]

Reviewer: ```suggestion
- The `changefeed.balance_range_distribution.enable` [cluster setting]({% link v24.1/cluster-settings.md %}) is now deprecated. Instead, use the new cluster setting `changefeed.default_range_distribution_strategy`. `changefeed.default_range_distribution_strategy='balanced_simple'` has the same effect as setting `changefeed.balance_range_distribution.enable=true`. It does not require `initial_scan='only'`, which was required by the old setting. [#115166][#115166]
```

- Added the [cluster setting]({% link v24.1/cluster-settings.md %}) `security.client_cert.subject_required.enabled` which enforces a mandatory requirement for the client certificate's role subject to be set. The subject can be defined through either the subject role option or by specifying the `root-cert-distinguished-name` and `node-cert-distinguished-name` properties. This setting applies to both RPC access and login via authCert. [#122368][#122368]

> - The [cluster setting]({% link v24.1/cluster-settings.md %}) `sql.contention.record_serialization_conflicts.enabled` is now `on` by default. This means any [`40001` errors]({% link v24.1/transaction-retry-error-reference.md %}) that are returned containing conflicting transaction information will be recorded by the contention registry, improving the ability to troubleshoot. For more information, refer to the [Insights page]({% link v24.1/ui-insights-page.md %}) documentation. [#116664][#116664]

Reviewer: ```suggestion
- The [cluster setting]({% link v24.1/cluster-settings.md %}) `sql.contention.record_serialization_conflicts.enabled` is now `on` by default. As a result, any [`40001` error]({% link v24.1/transaction-retry-error-reference.md %}) that contains conflicting transaction information will be recorded by the contention registry, improving the ability to troubleshoot. For more information, refer to the [Insights page]({% link v24.1/ui-insights-page.md %}) documentation. [#116664][#116664]
```

- The new [cluster setting]({% link v24.1/cluster-settings.md %}) `storage.sstable.compression_algorithm` configures the compression algorithm used when compressing sstable blocks. Supported values are: "snappy" and "zstd" [snappy = `1`, zstd = `2`]. Changing the default of snappy to zstd can result in substantial performance improvement, however, the effects this change may be highly dependent on the workload and data, so experimentation is recommended before enabling zstd in production environments.
- The new setting `storage.wal_failover.unhealthy_op_threshold` allows you to set the latency threshold at which a WAL (Write-Ahead Logging) write is considered unhealthy. When exceeded, the node will attempt to write WAL entries to a secondary store's volume. For more information, refer to <v24.1/cockroach-start.md#write-ahead-log-wal-failover> [#120509][#120509]

> - Added the `server.max_open_transactions_per_gateway` [cluster setting]({% link v24.1/cluster-settings.md %}). When set to a non-negative value, non-`admin` users cannot execute a query if the number of transactions open on the current gateway node is already at the configured limit. [#118781][#118781]

Reviewer: ```suggestion
- The new `server.max_open_transactions_per_gateway` [cluster setting]({% link v24.1/cluster-settings.md %}), when set to a non-negative value, allows only admin users to execute a query if the number of open transactions on the current gateway node is already at the configured limit. [#118781][#118781]
```

> - Introduced the `server.redact_sensitive_settings.enabled` [cluster setting](https://www.cockroachlabs.com/docs/v24.1/cluster-settings), which is false by default. If set to `true`, then the values of the following settings will be redacted when accessed through `SHOW` commands or other introspection interfaces. In the future, any other sensitive cluster settings that are added will be redacted as well. Users who have the `MODIFYCLUSTERSETTING` [privilege](https://www.cockroachlabs.com/docs/v24.1/security-reference/authorization#managing-privileges) can always view the unredacted settings. [#117729][#117729]

Reviewer: ```suggestion
- The new `server.redact_sensitive_settings.enabled` [cluster setting](https://www.cockroachlabs.com/docs/v24.1/cluster-settings) (`false` by default), when set to `true`, redacts the values of the following settings in the output of `SHOW` commands or other introspection interfaces. In the future, newly-added sensitive cluster settings will be redacted as well. Users with the `MODIFYCLUSTERSETTING` [privilege](https://www.cockroachlabs.com/docs/v24.1/security-reference/authorization#managing-privileges) can always view the unredacted settings. [#117729][#117729]
```
